mirrord/config/src/util.rs

```diff
@@ -35,11 +35,11 @@ where
 {
     type Generated = T::Generated;
 
-    fn generate_config(self) -> Result<Self::Generated, ConfigError> {
+    fn generate_config(self, warnings: &mut Vec<String>) -> Result<Self::Generated, ConfigError> {
```

Comment:
One little thing that will make future work easier on it is maybe wrapping the `Vec<String>` in some `ConfigContext` struct (like if maybe `String` should be `LogMessage` or have some extra data like a pretty error)

Reply:
Nice!